proto-lens/src/Data/ProtoLens/Encoding/Growing.hs

```diff
@@ -1,4 +1,16 @@
 -- | A mutable vector that grows in size.
+--
+-- Example usage:
+--
+-- > import qualified Data.ProtoLens.Encoding.Growing as Growing
+-- > import qualified Data.Vector.Unboxed as V
+-- > test :: IO (V.Vector Int)
+-- > test = do
+-- >     v <- Growing.new
+-- >     v' <- Growing.append v 1
+-- >     v'' <- Growing.append v' 2
+-- >     v''' <- Growing.append v'' 3
+-- >     unsafeFreeze v'''
```

Comment:
Come to think of it, does it make sense to turn this into a test? I know all this stuff is well covered by generated code that we test. It might still be useful to have a direct test considering how unsafe the innards are.

Reply:
Ah yeah, good point.  I intended to add unit tests for this package but forgot.  Will do.